### PR TITLE
Merge v1.4-rc1 back into master branch.

### PR DESCRIPTION
**Pull request by case540**
Mostly version string changes, but also adds some documentation that was directly committed
to r1.4 branch.

### CODE REVIEW

**Comment by case540** [comment]
This merge was pretty messy. For some reason I had tons of merge conflicts. Pretty sure I resolved everything correctly but please give it a glance over. Thanks!

File: tensorflow/docs_src/api_guides/cc/guide.md [hunk 1]
```diff
@@ -1,4 +1,12 @@
 # C++ API
+
```
Comment (av8ramit) [inline tensorflow/docs_src/api_guides/cc/guide.md]: Are changes to this doc intentional?

Comment (case540) [reply tensorflow/docs_src/api_guides/cc/guide.md]: Yes, this for instance was a doc fix merged directly into r1.4
https://github.com/tensorflow/tensorflow/pull/13858/files

**Review by av8ramit — APPROVED** [review]

**Comment by case540** [comment]
Weird, the bot is trying to build a target added a few days ago and can't find it. Going to try to trigger the test again to see if it works now.

**Comment by case540** [comment]
+gunan. Do you know what to do here. Ubuntu Makefile seems to be trying to build 
//tensorflow/contrib/makefile:ci_makefile_test

which is claims does not exist.

**Comment by benoitsteiner** [comment]
Jenkins, test this please.

**Comment by benoitsteiner** [comment]
I signed it

**Comment by benoitsteiner** [comment]
@gunan The cla bot is not running and prevents the merge. Assigning to you.

**Comment by gunan** [comment]
Mike could you resolve conflicts?

**Comment by case540** [comment]
Jenkins, test this please

**Comment by gunan** [comment]
Mike, could you look into the test failures?
Once they are addressed, I can merge.

Jenkins, test this please.

**Comment by case540** [comment]
The failures seem to be related to some Jenkins Github plugin. I dont see any actual test failures.

**Pull request merged**